Makefile

```diff
@@ -318,15 +339,18 @@ endif
 test-e2e-shared-minimal: build-all
 	mkdir -p "$(LOG_DIR)" "$(WORK_DIR)/.kcp"
 	rm -f "$(WORK_DIR)/.kcp/ready-to-test"
-	UNSAFE_E2E_HACK_DISABLE_ETCD_FSYNC=true NO_GORUN=1 \
-		./bin/test-server --quiet --log-file-path="$(LOG_DIR)/kcp.log" $(TEST_SERVER_ARGS) 2>&1 & PID=$$! && echo "PID $$PID" && \
-	trap 'kill -TERM $$PID' TERM INT EXIT && \
-	while [ ! -f "$(WORK_DIR)/.kcp/ready-to-test" ]; do sleep 1; done && \
-	echo 'Starting test(s)' && \
-	NO_GORUN=1 GOOS=$(OS) GOARCH=$(ARCH) \
-		$(GO_TEST) -race $(COUNT_ARG) $(PARALLELISM_ARG) $(WHAT) $(TEST_ARGS) \
-		-args --use-default-kcp-server $(SUITES_ARG) \
-	$(if $(value WAIT),|| { echo "Terminated with $$?"; wait "$$PID"; },)
+	USE_PROMETHEUS="$(USE_PROMETHEUS)" \
+		LOG_DIR="$(LOG_DIR)" \
+		WORK_DIR="$(WORK_DIR)" \
+		UNSAFE_E2E_HACK_DISABLE_ETCD_FSYNC=true \
+		NO_GORUN=1 \
+		GOOS=$(OS) \
+		GOARCH=$(ARCH) \
+		TEST_SERVER="./bin/test-server --quiet --log-file-path="$(LOG_DIR)/kcp.log" $(TEST_SERVER_ARGS)" \
+		TEST_CMD="$(GO_TEST) -race $(COUNT_ARG) $(PARALLELISM_ARG) $(WHAT) $(TEST_ARGS) -args \
+			--use-default-kcp-server \
+			$(SUITES_ARG)" \
+		hack/test-e2e.sh
```

Comment:
Rule(s) of thumb:
- these test targets run go test and the test server directly. No shell script indirection *)
- prometheus or any other testing setup beyond the test server and go test goes elsewhere, e.g. into prow+github-actions. These can share code via hack scripts or this `require-prometheus` target **)

(*) reason: with an indirection these test setups will grow more and more because it is to easy to add yet another thing, and another thing. Upper rule keeps us honest about test setup complexity.
(**) the primary user of these test targets is the developer. CI is secondary. Developer experience and simplicity matters. If there is a little complexity in the CI code around these targets that's fine.

Reply:
@sttts addressed, ptal